shared/actions/teams.js

```diff
@@ -1311,6 +1312,30 @@ const receivedBadgeState = (state: TypedState, action: NotificationsGen.Received
     })
   )
 
+const gregorPushState = (_: any, action: GregorGen.PushStatePayload) => {
+  const actions = []
+  const items = action.payload.state
+  const sawChatBanner = items.find(i => i.item && i.item.category === 'sawChatBanner')
+  if (sawChatBanner) {
+    actions.push(Saga.put(TeamsGen.createSetTeamSawChatBanner()))
+  }
+
+  const sawSubteamsBanner = items.find(i => i.item && i.item.category === 'sawSubteamsBanner')
+  if (sawSubteamsBanner) {
+    actions.push(Saga.put(TeamsGen.createSetTeamSawSubteamsBanner()))
+  }
+
+  const chosenChannels = items.find(i => i.item && i.item.category === Constants.chosenChannelsGregorKey)
+  const teamsWithChosenChannelsStr =
+    chosenChannels && chosenChannels.item && chosenChannels.item.body && chosenChannels.item.body.toString()
+  const teamsWithChosenChannels = teamsWithChosenChannelsStr
+    ? I.Set(JSON.parse(teamsWithChosenChannelsStr))
```

Comment:
worth worrying about whether JSON.parse throws and kills the other actions?

Reply:
no, this comes from the daemon so i assume its correctly formatted. this is just moved code so i don't want to touch it too much